src/shared/image-policy.c

```diff
@@ -0,0 +1,653 @@
+         * all needs no protection, because it *is* the protection */
+        if (partition_verity_to_data(policy->designator) >= 0 ||
+            partition_verity_sig_to_data(policy->designator) >= 0)
+                flags &= ~(PARTITION_POLICY_VERITY|PARTITION_POLICY_SIGNED|PARTITION_POLICY_ENCRYPTED);
+
+        /* if this designator has no verity concept, then mask off verity protection flags */
+        if (partition_verity_of(policy->designator) < 0)
+                flags &= ~(PARTITION_POLICY_VERITY|PARTITION_POLICY_SIGNED);
+
+        if ((flags & _PARTITION_POLICY_USE_MASK) == PARTITION_POLICY_ABSENT)
+                /* If the partition must be absent, then the gpt flags don't matter */
+                flags &= ~(_PARTITION_POLICY_READ_ONLY_MASK|_PARTITION_POLICY_GROWFS_MASK);
+        else {
+                /* If the gpt flags bits are not specified, set both options for each */
+                if ((flags & _PARTITION_POLICY_READ_ONLY_MASK) == 0)
+                        flags |= PARTITION_POLICY_READ_ONLY_ON|PARTITION_POLICY_READ_ONLY_OFF;
+                if ((flags & _PARTITION_POLICY_GROWFS_MASK) == 0)
+                        flags |= PARTITION_POLICY_GROWFS_ON|PARTITION_POLICY_GROWFS_OFF;
+        }
+
+        return flags;
+}
+
+PartitionPolicyFlags image_policy_get(const ImagePolicy *policy, PartitionDesignator designator) {
+        PartitionDesignator data_designator = _PARTITION_DESIGNATOR_INVALID;
+        PartitionPolicy *pp;
+
+        /* No policy means: everything may be used in any mode */
+        if (!policy)
+                return partition_policy_normalized_flags(
+                                &(const PartitionPolicy) {
+                                        .flags = PARTITION_POLICY_OPEN,
+                                        .designator = designator,
+                                });
+
+        pp = image_policy_bsearch(policy, designator);
+        if (pp)
+                return partition_policy_normalized_flags(pp);
+
+        /* Hmm, so this didn't work, then let's see if we can derive some policy from the underlying data
+         * partition in case of verity/signature partitions */
+
+        data_designator = partition_verity_to_data(designator);
+        if (data_designator >= 0) {
+                PartitionPolicyFlags data_flags;
+
+                /* So we are asked for the policy for a verity partition, and there's no explicit policy for
+                 * that case. Let's synthesize policy from the protection setting for the underlying data
+                 * partition. */
+
+                data_flags = image_policy_get(policy, data_designator);
+                if (data_flags < 0)
+                        return data_flags;
+
+                /* We need verity if verity or verity with sig is requested */
+                if (!(data_flags & (PARTITION_POLICY_SIGNED|PARTITION_POLICY_VERITY)))
+                        return _PARTITION_POLICY_FLAGS_INVALID;
```

Comment:
it just means "hey, no idea about that partition, i have no policy for it, nor can i synthesize one".

note that you are looking at image_policy_get() here. There's a wrapper for that image_policy_get_exhaustively() defined immediately after. If we return _PARTITION_POLICY_FLAGS_INVALID here it will convert that case into partition policy that is defined as default for unspeicified policies. So if you so will, returning _PARTITION_POLICY_FLAGS_INVALID here just means "default policy" (which by default means "ignore this partition")